src/model/gen.mc

```diff
@@ -1,102 +1,107 @@
-include "dfa.mc"
 include "string.mc"
 include "map.mc"
+include "dfa.mc"
 
 
 
--- Parsing all the accepted states
-recursive
-let parseAcceptedStates = lam aStates. lam output. lam map.
-    if(eqi (length aStates) 0) then output
-    else
-    let first = head aStates in
-    let rest = tail aStates in
-    let output = strJoin "" [output, "'", (int2string (mapLookup eqi first map)) , "',"] in
-    parseAcceptedStates rest output map
-end
 
 -- Formatting the states
 recursive
-let parseStates = lam states. lam startState. lam trans. lam map. lam output.
+let parseStates = lam states. lam startState. lam trans. lam dfa. lam output. 
     if (eqi (length states) 0) then output
     else
     let first = head states in
     let rest = tail states in
     let parsedFirst = strJoin "" [ "{id: '",
-    (int2string (mapLookup eqi first map)),
+    (int2string first.id),
     "', label:'",
-    (int2string first),
+    (dfa.s2s first.data),
     "'},\n"] in
-    parseStates rest startState trans map (concat output parsedFirst)
+    parseStates rest startState trans dfa (concat output parsedFirst)
 end
```

Comment:
Is it not more correct to call this function 'formatStates', as your comment suggest? There are a few more functions starting with 'parse' which should probably be changed to 'format'.